### DISCUSSION TITLE
UI Design suggestions

### DISCUSSION THREAD

**Discussion by SecularSteve** [sayaka-sh/spmp#189] [Ideas]
I'll keep my envisioned examples to the point by numbering the things I at least would like to see be optional with customisation in the preferences.

<img src="https://github.com/toasterofbread/spmp/assets/33793273/044d3f67-5b5c-4024-9a5f-59b75952df7b" width=50%>

1. The Radio Icon
2. The Like Icon
3. The Shuffle Button
4. The Repeat Button
5. The Top Queue Bar. To indicate and/or give a hint to the user that they can swipe up to bring up the Queue menu.

Speaking of which:

<img src="https://github.com/toasterofbread/spmp/assets/33793273/a86f55b7-66f7-4e42-b546-ca1f2d822a9a" width=50%>

(5.) The part of the queue I suggested to be visible in the Player menu.

6. A "Swipe" Column. Instead of having to drag your finger all the way up and swipe down to return to the Player menu, a Space on the left and/or right side of the Queue can be optionally turned on for the sole purpose of swiping the queue down without putting your finger all the way up. This could especially be useful for users with large vertical phone screens.

Any objections to these ideas? It's just a suggestion, not a demand.

**Comment by sayaka-sh** [comment]
Thanks for the suggestions! I've added no. 6 in 268d58355b4593b7122114a76a854830c441958d.

As for your other suggestions, I'm definitely open to redesigning the player layout (it was never really final in the first place). 

Two questions:
- What do you mean by radio button?
- An indicator for the queue is definitely something I could implement, but surely just showing the top of the queue would be too cluttered (and would also show the shuffle and repeat buttons twice if I also added them to the player like you suggested)

**Reply by SecularSteve** [reply]
Thanks. As for your questions:

1. The "Start Radio" button basically.
2. I expected this to come into discussion, and within reason, so I propose something like a basic arrow indicator instead:
https://www.svgrepo.com/show/349705/chevrons-down.svg

Simple, efficient and gets the point across.
But that's merely my suggestion.

**Comment by sayaka-sh** [comment]
I didn't use the same positions as your suggestion, but I added/moved the like, start radio, repeat, and shuffle buttons to the player in 4e2b174c01505a6e8a2efa98ffef8c699dad6156 81a253227f7d0795d9d33b8ac7d092f478db8c16.

Position 1: Repeat mode
Position 2: Shuffle
Position 3: Like/dislike
Position 4: Start radio

The repeat and shuffle buttons are hidden by default, and can be shown using an option in the player category.
I also moved the song details button to the bottom-right corner and added a button/indicator for opening the queue to the bottom-center.

Consider this a first draft, I'll probably tweak the layout a fair bit over the next few days/weeks. Feedback and/more more suggestions would be appreciated!

**Reply by SecularSteve** [reply]
Sounds good already.
I can't wait to try it out.

Btw, your implementation of "6." works like a charm.

**Reply by vichusachu** [reply]
Sir sorry for asking,is it possible to get the spmp apk without any hidden files ,i mean most of the drawables,strings,layouts etc are hidden and i like to customize your app,not for sharing but for my personal usage.if possible can u please share the app apk without any obfuscated files🤕,will be really helpfull 💖
